iframe example - use an onLoad callback instead of the ref

adrid opened the pull request:
**Description**
Makes the iframe example work on the Firefox and keeps it working on other browsers. It changes using the ref on the iframe to an `onLoad` event solution

**Issue**
Fixes: https://github.com/ianstormtaylor/slate/issues/4285

**Example**
https://www.slatejs.org/examples/iframe on the Firefox

**Context**
Firefox resets the iframe content on the load event. I've tested on some older versions of the browser and it keeps happening too. Here is an issue on the react repo https://github.com/facebook/react/issues/22847

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)

nemanja-tosic commented on site/examples/iframe.tsx:
```diff
@@ -104,15 +104,17 @@ const MarkButton = ({ format, icon }) => {
+  const [iframeBody, setIframeBody] = useState(null)
+
+  const handleLoad = e => {
+    onLoad && onLoad(e)
```
```suggestion
    onLoad?.(e)
```

adrid replied:
I've removed that part completely as not needed

nemanja-tosic commented on site/examples/iframe.tsx:
```diff
@@ -104,15 +104,17 @@ const MarkButton = ({ format, icon }) => {
-    contentRef &&
-    contentRef.contentWindow &&
-    contentRef.contentWindow.document.body
+const IFrame = ({ children, onLoad = null, ...props }) => {
```
Any specific reason to assign null to `onLoad`, instead of leaving it undefined?

```suggestion
const IFrame = ({ children, onLoad, ...props }) => {
```

adrid replied:
Thanks for pointing that out. I was experimenting with solutions and I had to left out it there. And actually that whole `onLoad` is not needed there... I will remove it

nemanja-tosic commented:
Just to make sure, it would not be possible to pass in `onLoad` load anymore. Is that intentional?

adrid commented:
> Just to make sure, it would not be possible to pass in `onLoad` load anymore. Is that intentional?

Yes. I don't think this is needed for this example.

dylans commented:
@adrid, this change has broken our integration test. Any chance you have time to update the test to reflect the change in the example?

```Running:  examples/iframe.ts                                                             (8 of 22)


  iframe editor
    1) should be editable


  0 passing (10s)
  1 failing

  1) iframe editor
       should be editable:
     TestingLibraryElementError: Timed out retrying after 4000ms: Unable to find an accessible element with the role "textbox"

There are no accessible roles. But there might be some inaccessible roles. If you wish to access them, then set the `hidden` option to `true`. Learn more about this here: https://testing-library.com/docs/dom-testing-library/api-queries#byrole
      at Object.getElementError (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:1764:17)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:2729:25)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:2701:24)
      at eval (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:2754:25)
      at baseCommandImpl (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:477:16)
      at commandImpl (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:480:40)
      at getValue (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:504:23)
      at resolveValue (http://localhost:3000/__cypress/tests?p=cypress/support/index.ts:544:35)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     10 seconds                                                                       │
  │ Spec Ran:     examples/iframe.ts                                                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

```

adrid commented:
@dylans when I created the pull request it was not passing this test and I noticed that later - the same thing what you pointing out. 

I already fixed it here in the next commit https://github.com/ianstormtaylor/slate/pull/4758/commits/87a860462051fbd45583e7404790727570b8ec29

I've tested on the main branch and seems to be okay so maybe you referring to some check before I fixed that?
